forge-game/src/main/java/forge/game/staticability/StaticAbilityDisableTriggers.java

```diff
@@ -0,0 +1,77 @@
+package forge.game.staticability;
+
+import com.google.common.collect.ImmutableList;
+
+import forge.game.Game;
+import forge.game.card.*;
+import forge.game.ability.AbilityKey;
+import forge.game.trigger.Trigger;
+import forge.game.trigger.TriggerType;
+import forge.game.zone.ZoneType;
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.util.Map;
+
+public class StaticAbilityDisableTriggers {
+
+    static String MODE = "DisableTriggers";
+
+    public static boolean disabled(final Game game, final TriggerType triggerType, final Trigger regtrig) {
+        //for AiController ETB tests
+        final Map<AbilityKey, Object> runParams = AbilityKey.mapFromCard(regtrig.getHostCard());
+        runParams.put(AbilityKey.Destination, ZoneType.Battlefield);
+        return disabled(game, triggerType, regtrig, runParams);
+    }
+
+    public static boolean disabled(final Game game, final TriggerType triggerType, final Trigger regtrig, final Map<AbilityKey, Object> runParams)  {
+        for (final Card ca : game.getCardsIn(ZoneType.STATIC_ABILITIES_SOURCE_ZONES)) {
+            for (final StaticAbility stAb : ca.getStaticAbilities()) {
+                if (!stAb.getParam("Mode").equals(MODE) || stAb.isSuppressed() || !stAb.checkConditions()) {
+                    continue;
+                }
+
+                if (isDisabled(stAb, triggerType, regtrig, runParams)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static boolean isDisabled(final StaticAbility stAb, final TriggerType triggerType, final Trigger regtrig, final Map<AbilityKey, Object> runParams) {
+        if (!stAb.matchesValidParam("ValidCard", regtrig.getHostCard())) {
+            return false;
+        }
+
+        if (stAb.hasParam("ValidMode")) {
+            if (!ArrayUtils.contains(stAb.getParam("ValidMode").split(","), regtrig.getMode().toString())) {
+                return false;
+            }
+        }
+
+        if (triggerType.equals(TriggerType.ChangesZone)) {
+            // Cause of the trigger – the card changing zones
+            if (!stAb.matchesValidParam("ValidCause", runParams.get(AbilityKey.Card))) {
```

Comment:
Looks like you didn't transfer the exception from Hushbringer yet:
> An ability that triggers when a card is put into a graveyard “from anywhere” may still trigger even if a creature dies. It will trigger if the ability is of a dying creature that triggers only when that creature is put into a graveyard from anywhere